float32 hue
float32 saturation
float32 value
